### PR TITLE
Support for tag validation before add

### PR DESCRIPTION
**Pull request by axelniklasson**
Makes it possible to validate a tag based on user-supplied validation function before adding it to the tags set. It simply checks if validation function is supplied and only calls `handleAddition` if said validation function returns true.

Let me know if something's missing and thanks for a great lib!

### CODE REVIEW

File: spec/helpers/dom.js [hunk 1]
```diff
@@ -6,7 +6,9 @@
 
 const jsdom = require('jsdom')
 
-const { window } = new jsdom.JSDOM('<html><body><div id="app"></div></body></html>', {})
+const { window } = new jsdom.JSDOM('<html><body><div id="app"></div></body></html>', {
+  url: "http://localhost/"
```
Comment (i-like-robots) [inline spec/helpers/dom.js]: 👌

**Comment by i-like-robots** [comment]
@axelniklasson Thank you for taking the time to create this issue but I'm a little confused by it. What is the advantage of handling validation inside this component rather than within the `handleAddition` callback supplied to it? I'm also wondering that if validation were to be handled within this component then there should also be a mechanism for which feedback can be given to the user in the event that their input is invalid.

**Comment by axelniklasson** [comment]
@i-like-robots No problem, thanks for a great lib! My issue with handling the validation in the `handleAddition` callback was that if I did not add the tag to the set of tags because of invalid data, the input text got cleared out. Maybe I used it wrong though? 

I definitely agree that there should be functionality to display some sort of error message as well, that's a great idea. 

I'm unfortunately a bit short on time to set up a fiddle, but can try to do it this weekend.

**Comment by i-like-robots** [comment]
Thank you @axelniklasson for this, and sorry about the delay. I will release this shortly under minor version number.

**Comment by axelniklasson** [comment]
Oh, I forgot about this one actually! Excited to see it made its way into master 😃

**Pull request merged**